### PR TITLE
Allow protected cookies to be configurable

### PR DESCRIPTION
**Pull request by mikefrancis**
Implements #37

### CODE REVIEW

File: tests/ApplySecureHeadersTest.php [hunk 1]
```diff
@@ -300,65 +268,27 @@ public function testCSPRO()
...
-        $secureHeaders->errorReporting(false);
-        $middleware = new ApplySecureHeaders($config, $secureHeaders);
+        $result  = $this->applySecureHeadersWithConfig(new Response, $configMap);
+        $headers = $result->headers->all();
 
-        return $middleware->handle(new Request, function () use ($response) {
-            return $response;
-        });
-    }
+        dd($result);
```
Comment (mikefrancis) [inline tests/ApplySecureHeadersTest.php:290]: @aidantwoods I'm not seeing anything in this dump after using `protectedCookie`:

https://github.com/mikefrancis/laravel-secureheaders/pull/38/files#diff-d7aa6dfd3694cac6b7ce2fe2b35c3a8cR134

Am I doing something wrong?

Comment (aidantwoods) [reply tests/ApplySecureHeadersTest.php:290]: Is the cookie set anywhere? :) The purpose of `protectedCookie` behaviour is to make sure the named cookies don't go out without flags like `secure`, but if the cookie is never set then there's nothing to add the flags to.

If you set cookies `lol` and `whut` (e.g. add the headers `Set-Cookie: lol=foo`, `Set-Cookie: whut=bar`) then the configured flags should be added.

Comment (mikefrancis) [reply tests/ApplySecureHeadersTest.php:290]: There's nothing being set, no. Not sure if this something to do with how the test applies the middleware to the request in `applySecureHeadersWithConfig()` 🤔

Comment (aidantwoods) [reply tests/ApplySecureHeadersTest.php:290]: You need to set some cookies that will match to observe an effect :) If no cookies are set then nothing happens.

E.g. if the cookie named `foo` is configured as a protected cookie and the following header is in the original response

```
Set-Cookie: foo=bar
```

then after applying SecureHeaders you should instead see the header:

```
Set-Cookie: foo=bar; Secure; HttpOnly; SameSite=Lax
```

Comment (mikefrancis) [reply tests/ApplySecureHeadersTest.php:290]: Of course 🤦 sorry, had a long night last night!

File: src/ApplySecureHeaders.php [hunk 2]
```diff
@@ -132,4 +125,35 @@ private function setExpectCT()
...
+    /**
+     * Set any protected cookies.
+     */
+    private function setProtectedCookies()
+    {
+        $this->headers->protectedCookie('lol');
+        // $namesToAdd = $this->config->get('secure-headers.protectedCookies.add.names', []);
+
+        // if (count($namesToAdd)) {
+        // $this->headers->protectedCookie($namesToAdd);
```
Comment (aidantwoods) [inline src/ApplySecureHeaders.php:138]: For exact names, make sure to pass the [`SecureHeaders::COOKIE_NAME`](https://github.com/aidantwoods/SecureHeaders/wiki/protectedCookie#cookie_name) constant in the second arg (and similar when removing names, but combined (bitwise OR) with `SecureHeaders::COOKIE_REMOVE`).

I think when I get around to v3 I'll remove all this bitwise stuff, not terribly straightforward compared to e.g. `addProtectedCookieSubstring`, `removeProtectedCookieSubstring` :)

Comment (mikefrancis) [reply src/ApplySecureHeaders.php:138]: Will do, this is very much WIP - just wanted to get a bit of help on the set cookie "issue"

**Pull request closed without merging**